test/Feature/AuthSpec.hs

```diff
@@ -31,15 +31,36 @@ spec = beforeAll
     request methodGet "/authors_only" [auth] ""
       `shouldRespondWith` 200
 
+  it "works with tokens which have extra fields" $ do
+    let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoicG9zdGdyZXN0X3Rlc3RfYXV0aG9yIiwiaWQiOiJqZG9lIiwia2V5MSI6InZhbHVlMSIsImtleTIiOiJ2YWx1ZTIiLCJrZXkzIjoidmFsdWUzIiwiYSI6MSwiYiI6MiwiYyI6M30.GfydCh-F4wnM379xs0n1zUgalwJIsb6YoBapCo8HlFk"
```

Comment:
The two options we have for this would be our `tokenJWT` method, which doesn't allow us to set any registered claims (such as `exp`). Or we use `Web.JWT.encodeSigned` which requires some ugly object declaration. Doable, but it's much more streamlined to define the JWT in code and just use `https://jwt.io` to edit it. Another latent function of keeping it in code is when debugging the server you can just copy/paste it to a header field.

I could do it, but it might be a little non-intuitive.